mlflow/tensorflow/__init__.py

```diff
@@ -1257,6 +1265,29 @@ def __restore_generator(prev_generator):
                 if log_models:
                     _log_keras_model(history, args)
 
+                if log_datasets:
+                    try:
+                        # create a CodeDatasetSource
+                        context_tags = context_registry.resolve_tags()
+                        source = CodeDatasetSource(
+                            mlflow_source_type=context_tags[MLFLOW_SOURCE_TYPE],
+                            mlflow_source_name=context_tags[MLFLOW_SOURCE_NAME],
+                        )
+
+                        training_data = kwargs["x"] if "x" in kwargs else args[0]
```

Comment:
What about logging the validation dataset, if specified as a kwarg? This would be really nice :)